.github/workflows/main.yml

```diff
@@ -50,7 +53,33 @@ jobs:
           flake8 . --exit-zero --max-complexity=6
 
       - name: Upload python artifacts
+        if: ${{ github.ref_name }} == 'main'
```

Comment:
The `if` condition is checking `github.ref_name` directly against 'main'. Ensure that the `github.ref_name` is correctly set to 'main' when this condition is evaluated. Consider using `github.ref` or `github.head_ref` if the condition is not behaving as expected.